front/src/Phaser/Player/Player.ts

```diff
@@ -94,39 +85,16 @@ export class Player extends Character {
         return this.wasMoving;
     }
 
-    openOrCloseEmoteMenu(emotes: RadialMenuItem[]) {
-        if (this.emoteMenu) {
-            this.closeEmoteMenu();
-        } else {
-            this.openEmoteMenu(emotes);
-        }
-    }
-
-    openEmoteMenu(emotes: RadialMenuItem[]): void {
-        this.cancelPreviousEmote();
-        this.emoteMenu = new RadialMenu(this.scene, this.x, this.y, emotes);
-        this.emoteMenu.on(RadialMenuClickEvent, (item: RadialMenuItem) => {
-            this.closeEmoteMenu();
-            this.emit(requestEmoteEventName, item.name);
-            this.playEmote(item.name);
-        });
-    }
-
-    isSilent() {
-        super.isSilent();
-    }
-    noSilent() {
-        super.noSilent();
-    }
-
-    closeEmoteMenu(): void {
-        if (!this.emoteMenu) return;
-        this.emoteMenu.destroy();
-        this.emoteMenu = null;
+    playEmote(emote: string) {
+        super.playEmote(emote);
+        emoteMenuStore.set(false);
     }
 
-    destroy() {
-        this.scene.events.removeListener("postupdate", this.updateListener);
-        super.destroy();
+    openOrCloseEmoteMenu() {
```

Comment:
I would take this method out of the class Player since it doesn't really interact with the phaser object. You could put it directly inside the emoteMenuStore or at least in EmoteStore.ts

Reply:
yes make sense, was just adopted from old code.